saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilter.java

```diff
@@ -74,23 +78,42 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 		}
 
 		String registrationId = matcher.getVariables().get("registrationId");
-		sendRedirect(request, response, registrationId);
+		RelyingPartyRegistration relyingParty = this.relyingPartyRegistrationRepository.findByRegistrationId(registrationId);
+		if (relyingParty == null) {
+			response.sendError(HttpServletResponse.SC_NOT_FOUND);
```

Comment:
I believe this should be a 401 to mitigate enumeration. Further, I can't seem to find a place in Spring Security that returns a 404 in similar circumstances.

Reply:
Adjusted. e11e8007bc356efd59f7c82f04a028e1ca305229